src/vm/parser.rs

```diff
@@ -70,9 +71,10 @@ pub fn lex(input: &str) -> Result<Vec<(LexItem, u32, u32)>> {
         LexMatcher::new("[(]", TokenType::LParens),
         LexMatcher::new("[)]", TokenType::RParens),
         LexMatcher::new("0x(?P<value>[[:xdigit:]]+)", TokenType::HexStringLiteral),
-        LexMatcher::new("(?P<value>[[:digit:]]+)", TokenType::IntLiteral),
+        LexMatcher::new("(?P<value>-?[[:digit:]]+)", TokenType::IntLiteral),
         LexMatcher::new("'(?P<value>true|false)", TokenType::QuoteLiteral),
-        LexMatcher::new("'CT(?P<value>[[:alpha:]]{5,40})", TokenType::ContractPrincipalLiteral),
+        LexMatcher::new("'CT(?P<value>[0123456789ABCDEFGHJKMNPQRSTVWXYZ]{28,41}.([[:alpha:]]|[-]){5,40})", TokenType::QualifiedContractPrincipalLiteral),
```

Comment:
I didn't see this before.  To be clear -- the minimum length of a contract name is 5 characters, and the maximum name is 40?  Is this true everywhere in the code, and is this long enough?  Asking because filenames are usually up to 256 characters on most UNIX systems, and I'm pretty sure most programming languages offer higher limits (if any at all).

Reply:
This limit is not otherwise enforced -- I think doing so should be done via a contract type (similar to the other comment on contract names). I don't see much reason for limiting the length of the contract names, so would be happy to change this to a `+`

Reply:
Is this added to an issue somewhere?

Reply:
I added a note at the bottom here: https://github.com/blockstack/blockstack-core/issues/1078